pkgs/by-name/fr/frankenphp/package.nix

```diff
@@ -0,0 +1,94 @@
+{ lib
+, stdenv
+, buildGoModule
+, fetchFromGitHub
+, php
+, pcre2
+, libargon2
+, libxml2
+, systemd
+, testers
+, frankenphp
+, runCommand
+, writeText
+
+, phpOverrideArgs ? {}
+}:
+
+let
+  phpEmbedWithZts = php.override {
+    embedSupport = true;
+    ztsSupport = true;
+  } // (lib.filterAttrs (n: v: !(builtins.elem n [ "embedSupport" "ztsSupport" ])) phpOverrideArgs);
+  phpUnwrapped = phpEmbedWithZts.unwrapped;
+  phpConfig = "${phpUnwrapped.dev}/bin/php-config";
+  pieBuild = stdenv.hostPlatform.isMusl;
+in buildGoModule rec {
+  pname = "frankenphp";
+  version = "1.0.0-rc.3";
+
+  src = fetchFromGitHub {
+    owner = "dunglas";
+    repo = "frankenphp";
+    rev = "v${version}";
+    hash = "sha256-Al0gCxTb6s41ugX9J8N8lshop9kP3RPGCzlq5etk1RY=";
+  };
+
+  sourceRoot = "source/caddy";
+
+  # frankenphp requires C code that would be removed with `go mod tidy`
+  # https://github.com/golang/go/issues/26366
+  proxyVendor = true;
+  vendorHash = "sha256-Lgj/pFtSQIgjrycajJ1zNY3ytvArmuk0E3IjsAzsNdM=";
+
+  buildInputs = [ phpUnwrapped pcre2 libargon2 libxml2 systemd ];
```

Comment:
`libxml` and `systemd` are only required because of `php` configuration. Ideally here I need to collect php `buildInputs` to make sure frakenphp will include dependencies that php need.

Reply:
Fixed in https://github.com/NixOS/nixpkgs/pull/268665/commits/e2c0798e29262d2bb21d56af8e7002607d591790